File: src/leetcode/python/ReconstructOriginalDigitsFromEnglish_423.py
class ReconstructOriginalDigitsFromEnglish_423(object):
    def originalDigits(self, s):
        """
        :type s: str
        :rtype: str
        """
        cnt = [0] * 10
        for c in s:
            if c == 'z': cnt[0] += 1
            elif c == 'w': cnt[2] += 1
            elif c == 'u': cnt[4] += 1
            elif c == 'x': cnt[6] += 1
            elif c == 'g': cnt[8] += 1
            elif c == 's': cnt[7] += 1
            elif c == 'f': cnt[5] += 1
            elif c == 'h': cnt[3] += 1
            elif c == 'o': cnt[1] += 1
            elif c == 'i': cnt[9] += 1
        cnt[3] -= cnt[8]
        cnt[5] -= cnt[4]
        cnt[7] -= cnt[6]
        cnt[9] -= (cnt[8] + cnt[6] + cnt[5])
        cnt[1] -= (cnt[2] + cnt[4] + cnt[0])
        return "".join(str(i)*c for i, c in enumerate(cnt))

File: src/leetcode/java/ReconstructOriginalDigitsFromEnglish_423.java
package leetcode.java;

public class ReconstructOriginalDigitsFromEnglish_423 {
  public String originalDigits(String s) {
    int[] counter = new int[10];
    for (char c : s.toCharArray()) {
      switch (c) {
        case 'z': counter[0]++; break;
        case 'w': counter[2]++; break;
        case 'u': counter[4]++; break;
        case 'x': counter[6]++; break;
        case 'g': counter[8]++; break;
        case 's': counter[7]++; break; // c[7] - c[6]
        case 'f': counter[5]++; break; // c[5] - c[4]
        case 'h': counter[3]++; break; // c[3] - c[8]
        case 'o': counter[1]++; break; // c[1] - c[0] - c[2] - c[4]
        case 'i': counter[9]++; break; // c[9] - c[8] - c[6] - c[5]
      }
    }
    counter[7] -= counter[6];
    counter[5] -= counter[4];
    counter[3] -= counter[8];
    counter[1] -= (counter[0] + counter[2] + counter[4]);
    counter[9] -= (counter[8] + counter[6] + counter[5]);
    StringBuilder builder = new StringBuilder();
    for (int i = 0; i < 10; i++) {
      for (int j = 0; j < counter[i]; j++) {
        builder.append(i);
      }
    }
    return builder.toString();
  }
}
